lib/ldap.js

```diff
@@ -42,7 +47,12 @@ function initializeConnection () {
       res.once('error', function(err) {
         client.removeAllListeners('end');
         clearTimeout(abort);
-        callback(err);
+        // if there are more than one entry matching the search, the server returns the one entry and a SizeLimitExceededError error
+        if (err.name === 'SizeLimitExceededError') {
```

Comment:
For Anonymous Search detection logic, this approach is fine because we execute it once.

For heartbeat, I'd like to avoid errors (`(objectclass=person)` will always return more than 1 entry, so the error event will be always triggered).

Our plan is to replace `(objectclass=*)` with a valid query that returns no entries (or a small number of entries). Something like `(&(objectclass=user)(|(sAMAccountName=foo)(UserPrincipalName=foo)))`.

WDYT?

Reply:
If I understand this correctly, `SizeLimitExceededError` is just telling us that the results returned by the search would be exceeding the configured heartbeat search `sizeLimit`? 

In which case I feel that in this case treating the `SizeLimitExceededError` as an "expected scenario" would have no harm.

Potentially, replacing the open search query with something more specific sounds good to me, if we could do that robustly with no assumptions about expected AD contents.

Thinking of this mechanism similarly to the `LIMIT N` clause on an SQL query, I'd be comfortable with the additional safety-guard limiting the `sizeLimit`/`scope` of the query.

Ideally this heartbeat should just be a "healthchecking" search query right? Is there anything fit-for-porpuse in LDAP we could rely on?

Reply:
> If I understand this correctly, SizeLimitExceededError is just telling us that the results returned by the search would be exceeding the configured heartbeat search sizeLimit?

Yes, that's correct. But, as I said before, for this particular case ("healthchecking"), it feels a little odd having to force an error (`SizeLimitExceededError`) to check if service is up and running.

That's it, both options are completely valid. But I'd prefer to avoid the noise from LDAP side (it will start generating logs for this error) and keep implementation simple in the connector.

Reply:
> But, as I said before, for this particular case ("healthchecking"), it feels a little odd having to force an error (SizeLimitExceededError) to check if service is up and running.

Agreed @siacomuzzi, would be nice to have a clause similar to the SQL `LIMIT N` one that doesn't generate errors server side, for this.

>  But I'd prefer to avoid the noise from LDAP side (it will start generating logs for this error) and keep implementation simple in the connector.

👍 sounds good to us, feel free to close this pr. We'll keep running this branch until there's a release with a simple search query, then switch back to upstream.

Thanks!

Reply:
We'll let you know once the release is ready.
Thanks for your help and reporting us!